auth/oidc.go

```diff
@@ -141,65 +174,58 @@ func (op *OIDCProvider) InitOIDCClient() error {
 		return base.RedactErrorf("Issuer not defined for OpenID Connect provider %+v", base.UD(op))
 	}
 
-	config, shouldSyncConfig, err := op.DiscoverConfig()
-	if err != nil || config == nil {
+	ctx := context.Background()
+	provider, _, err := op.DiscoverConfig(ctx)
```

Comment:
Using the local variable name 'providerConfig' here would be less confusing - currently it sounds like the same thing as OIDCProvider.